src/main/java/com/peerrequest/app/api/DirectRequestsController.java

```diff
@@ -0,0 +1,264 @@
+        if (option.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "direct request does not exist");
+        }
+
+        String researcherId = this.entryService.get(entryId).get().getResearcherId();
+        String reviewerId = option.get().getReviewerId();
+        if (!user.getAttribute("sub").toString().equals(researcherId)
+                || user.getAttribute("sub").toString().equals(reviewerId)) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN,
+                    "Only the researcher or reviewer may get the request");
+        }
+        return option.get().toDto();
+    }
+
+    /**
+     * Gets the list of all requests of an entry.
+     *
+     * @return directRequest with requestId
+     */
+    @GetMapping(value = "/categories/{category_id}/entries/{entry_id}/process/requests",
+            produces = "application/json")
+    public List<DirectRequest.Dto> listDirectRequestsByEntry(@RequestParam Optional<Integer> limit,
+                                                             @RequestParam Optional<Long> after,
+                                                             @PathVariable("entry_id") final Long entryId,
+                                                             @AuthenticationPrincipal OAuth2User user) {
+        if (this.entryService.get(entryId).isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "entry does not exist");
+        }
+
+        String researcherId = this.entryService.get(entryId).get().getResearcherId();
+
+        if (!user.getAttribute("sub").toString().equals(researcherId)) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN,
+                    "Only the user that created the entry may get all requests of the entry");
+        }
+
+        if (limit.isPresent()) {
+            if (limit.get() < 0) {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "limit must be greater than 0");
+            }
+            limit = Optional.of(Math.min(limit.get(), maxPageSize));
+        }
+
+        var option = this.directRequestProcessService.get(entryId);
+        if (option.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "direct request process does not exist");
+        }
+
+        DirectRequest filterDirectRequest = new DirectRequest(null, null, null, option.get().getId());
+
+        return this.directRequestService.list(after.orElse(null), limit.orElse(maxPageSize),
+                        filterDirectRequest.toDto()).stream().map(DirectRequest::toDto).toList();
+    }
+
+    /**
+     * Patches a directRequest.
+     *
+     * @param requestId DirectRequest to patch
+     * @param updater updater containing a DirectRequest.State
+     *
+     * @return the updated request
+     */
+    @PatchMapping(value = "/categories/{category_id}/entries/{entry_id}/process/requests/{request_id}",
```

Comment:
Wrong endpoint. remove /{request_id} and get the id of the request from the requestBody.
It works like this too but we did it already different in the other controllers

Reply:
Done.